Individual Kudo Re-design

nnsookwon opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

### Code Review Deadline: EOD Tuesday 5/22
<!--- Please provide a reasonable code review deadline above -->
<!--- Follow any guidelines for setting CR deadlines if applicable -->

## This PR...
- Implements new design for individual Kudos
- Removes `font-awesome-rails` and uses a CDN link for the newest version of font awesome (v5)
- Bumps up `material-ui` package version (to make use of `<Tooltip />`

## How Has This Been Tested?
👀

## Media
![pretty-kudo-demo](https://user-images.githubusercontent.com/20546812/40210113-062ebc82-59f8-11e8-8dd8-cedf84ab652c.gif)

![may-17-2018 17-31-20](https://user-images.githubusercontent.com/20546812/40210150-31000bd2-59f8-11e8-890d-6cf99fd8f423.gif)


<img width="626" alt="screen shot 2018-05-17 at 3 55 56 pm" src="https://user-images.githubusercontent.com/20546812/40210139-28a1f5b8-59f8-11e8-9561-6866e81ccfe4.png">




## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Refactor (non-breaking update to code that doesn't change functionality or external behavior, may include some new code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- For all the points that should be checked but aren't please explain why after the list. -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Mandatory Reviewer: @trisarahsorus 
## Secondary Reviewer: @joecox
 (Tagging you so you see, but don't worry about the deadline date)
<!--- Autocomplete the mandatory reviewer’s username after the @ symbol above. -->

nnsookwon commented on client/app/bundles/KudosApp/components/KudosList.jsx:
```diff
@@ -1,32 +1,41 @@
+      kudos.map((kudo, index) => (
+        <Kudo
+          id={ kudo.id }
+          color_class={ color_classes[(numKudos-index)%numColors] }
```
FYI I did this so that when a new Kudo is submitted (and inserted at the top AKA becomes the first item in the array) this won't affect the colors of the existing Kudos on the screen.

Right now it works, but after adding some kind of scroll to load more kudos, we'll need a different method of assigning colors since kudos can be appended at the beginning and end

nnsookwon replied:
Thanks! I'll keep this is mind as I try to add scroll load feature next

nnsookwon commented on client/app/bundles/KudosApp/components/Kudo.jsx:
```diff
@@ -1,127 +1,175 @@
-    let poster = this.props.kudo.giver_id;
-    return (user === poster);
+  formattedTimestamp() {
+    const ts = moment(this.props.kudo.given_at)
```
Cool! I'll try replacing `moment` with `dayjs`

nnsookwon commented on client/app/bundles/KudosApp/components/Kudo.jsx:
```diff
@@ -1,127 +1,175 @@
-    this.setState({editing: true});
+  formattedLikeText() {
+    const { likes } = this.props.kudo
+    const numLikes = isEmpty(likes) ? 0 : likes.length
```
This was a copy-paste, but you're right! Maybe it was a precaution in case `likes` might be undefined, but it seems that each kudo should have a `likes` attribute defined as an empty array even if there's no likes. Thanks!

joecox commented on client/app/bundles/KudosApp/components/Kudo.jsx:
```diff
@@ -1,127 +1,175 @@
+        <div className="meta">
+          <div>
+            { this.renderLikeIcon() }
+            { this.formattedLikeText() }
```
Let's just remove the extra like copy altogether.

nnsookwon replied:
Do you mean remove the text following the number of likes? e.g. "1 person likes this"

joecox replied:
Yeah, I'm thinking only numbers is fine, what do you think?

joecox commented on client/app/bundles/KudosApp/components/KudosList.jsx:
```diff
@@ -69,7 +78,7 @@ export default class KudosList extends React.Component {
       <TabBarContainer />
-      <List giverId={this.props.id} kudos={this.props.kudos} likeKudo={this.props.likeKudo} unlikeKudo={this.props.unlikeKudo} updateKudo={this.props.updateKudo} />
+      <List userId={this.props.id} kudos={this.props.kudos} likeKudo={this.props.likeKudo} unlikeKudo={this.props.unlikeKudo} updateKudo={this.props.updateKudo} />
       {this.props.isFetchingKudos ? <Spinner /> : null}
```
I noticed on prod too that the `List` will say 'No Kudos' even while the spinner is going.  Can we change these two lines to be:
```js
this.props.isFetchingKudos ? <Spinner /> : <List ... />
```

nnsookwon replied:
Instead I decided to change the logic for display 'No kudos' only if its length 0 AND it's not fetching (which should indicate that a fetch has been attempted already).

I did this because the spinner will be probably be able to be re-used for scroll-loading, and I wouldn't want the entire list of currently loaded kudos to disappear while fetching additional kudos.

joecox replied:
^oh yeah I forgot about that, good call

joecox commented:
What I'm referencing in my other comment:
<img width="669" alt="screen shot 2018-05-22 at 4 34 53 pm" src="https://user-images.githubusercontent.com/2593686/40395742-61f047a4-5dde-11e8-9d49-c14b788cb515.png">

nnsookwon commented:
@trisarahsorus's and @joecox's requests have been added!

Also, @joecox I believe the kudos expanding issues is fixed with a CSS attribute I added for the two-column feature. But still, I've added the default avatar image so it looks perty

<img width="618" alt="screen shot 2018-05-23 at 2 38 18 pm" src="https://user-images.githubusercontent.com/20546812/40452614-49a84cb4-5e97-11e8-9f85-1290086d1231.png">
